### PR TITLE
add backend list to landing page

### PR DESCRIPTION
**Pull request by CalMacCQ**
Added the list of available QPUs/ simualtors to the landing page of the extensions docs. I have also removed the list from the README so we don't have to update this in multiple places.

### CODE REVIEW

File: .github/workflows/docs/intro.txt [hunk 1]
```diff
@@ -6,10 +6,119 @@ pytket-extensions
...
+* Unitary simulators - Unitary simulators calculate the unitary operation that is applied by a circuit. A unitary matrix/ndarray is returned ``AerUnitaryBackend`` is an example of such a simulator.
+* Density matrix simulators - These simulators compute the density matrix prepared by a circuit. The result can be a statistical mixture of states in contrast to statevector simulation. E.g. ``CirqDensityMatrixSampleBackend`` 
+* Other specialised simulators - There are extensions for simulating specific types of circuit. For instance the ``SimplexBackend`` is designed to simulate Clifford circuits. 
+
+QPUs
+====
+
+`IBMQBackend <https://cqcl.github.io/pytket-extensions/api/qiskit/api.html#pytket.extensions.qiskit.IBMQBackend>`_
+A backend for running circuits on remote IBMQ devices.
+'
```
Comment (cqc-melf) [inline .github/workflows/docs/intro.txt:26]: I think you can remove this ' it is not needed.

**Review by cqc-melf — CHANGES_REQUESTED** [review]

**Comment by cqc-melf** [comment]
We should update the links to the quantinuum docs which are soon at https://cqcl.github.io/pytket-quantinuum/api/

File: .github/workflows/docs/intro.txt [hunk 2]
```diff
@@ -6,10 +6,123 @@ pytket-extensions
...
-platforms. Each extension adds either new methods to the ``pytket`` package to
-convert between circuit representations, or new backends to which ``pytket``
-circuits can be submitted.
+These extensions enable ``pytket`` to interface with different hardware platforms and classical simulators. One can also access additional quantum devices and simulators via the cloud through the extensions for `Azure <https://cqcl.github.io/pytket-extensions/api/qsharp/api.html#pytket.extensions.qsharp.AzureBackend>`_ and `Braket <https://cqcl.github.io/pytket-extensions/api/braket/api.html#pytket.extensions.braket.BraketBackend>`_ . 
+In pytket a ``Backend`` represents a connection to a QPU (Quantum Processing Unit) or simulator for processing quantum circuits. The types of ``Backend`` available in pytket are the following.
+
+Types of Backend
+================
+
+* **QPUs** - These are real quantum computers that return shots based results. E.g the `QuantinuumBackend <https://cqcl.github.io/pytket-quantinuum/api/>`_ .
```
Comment (cqc-melf) [inline .github/workflows/docs/intro.txt:15]: The right link should be cqcl.github.io/pytket-quantinuum/api/api.html#pytket.extensions.quantinuum.QuantinuumBackend

Comment (CalMacCQ) [reply .github/workflows/docs/intro.txt:15]: Thanks
Should be correct now.

**Review by cqc-melf — APPROVED** [review]
Looks good, thank you for doing this!

**Pull request merged**